server/ServerApp.cpp

```diff
@@ -1638,6 +1638,41 @@ void ServerApp::AddObserverPlayerIntoGame(const PlayerConnectionPtr& player_conn
     // TODO: notify other players
 }
 
+void ServerApp::DropPlayerEmpireLink(int player_id)
+{ m_player_empire_ids.erase(player_id); }
+
+bool ServerApp::AddPlayerIntoGame(const PlayerConnectionPtr& player_connection) {
+    // search empire by player name
+    for (auto empire : Empires()) {
+        if (empire.second->PlayerName() == player_connection->PlayerName()) {
+            auto orders_it = m_turn_sequence.find(empire.first);
+            if (orders_it == m_turn_sequence.end()) {
+                WarnLogger() << "ServerApp::AddPlayerIntoGame empire " << empire.first
+                             << " for \"" << player_connection->PlayerName()
+                             << "\" doesn't wait for orders";
+                return false;
+            }
+            // make a link to new connection
+            m_player_empire_ids[player_connection->PlayerID()] = empire.first;
+
+            const OrderSet& orders = orders_it->second ? *orders_it->second : OrderSet();
```

Comment:
I think you're binding a reference to a temporary here...

Reply:
Fixed.